eclipse/che-theia
# Fix displaying output for vs code tasks

### What does this PR do?
Output for `vs code`/`theia` tasks is not displayed after changes in upstream.
The cause of the problem: a mechanism of creating terminal for a task was changed on `theia` side.

The current PR fixes the issue and allows to manage what kind of terminal widget is created for a task (theia terminal widget vs remote terminal widget)

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/16331

### How to test?
You can use the following devfile to test the changes 

<details>
<summary>Devfile</summary>

```
apiVersion: 1.0.0
metadata:
  name: test-theia-tasks
projects:
  - name: che-theia
    source:
      type: git
      location: 'https://github.com/eclipse/che-theia.git'
  - name: theia
    source:
      type: git
      location: 'https://github.com/eclipse-theia/theia.git'
components:
  - alias: che-dev
    type: dockerimage
    image: 'quay.io/eclipse/che-theia-dev:next'
    mountSources: true
    endpoints:
      - name: theia-dev-flow
        port: 3010
        attributes:
          protocol: http
          public: 'true'
    memoryLimit: 3Gi
  - id: redhat/vscode-yaml/latest
    type: chePlugin
  - id: che-incubator/typescript/latest
    type: chePlugin
    memoryLimit: 2048M
  - type: cheEditor
    alias: theia-editor
    reference: https://raw.githubusercontent.com/RomanNikitenko/che-plugin-registry/master/v3/plugins/eclipse/che-theia/fix-theia-tasks/meta.yaml
    memoryLimit: 2Gi
commands:
  - name: 'che:theia:init'
    actions:
      - type: exec
        component: che-dev
        command: |
          che-theia init
        workdir: /projects/theia
  - name: 'theia:build'
    actions:
      - type: exec
        component: che-dev
        command: |
          yarn
        workdir: /projects/theia
  - name: 'che-theia:build'
    actions:
      - type: exec
        component: che-dev
        command: |
          yarn
        workdir: /projects/che-theia



```
</details>

The image for `theia-editor` component based on:
- current master `theia` branch
- current PR branch for `che-theia`

More details you can get using `Help -> About` dialog.

**Steps for testing:**
1. Run `vs code` tasks (`npm`, `typescript`) using `Terminal` -> `Run Task` menu
2. Run a `che` task using `Terminal` -> `Run Task` menu
3. Run a `che` task using `My Workspace` panel 
4. Try to add and run a `shell` task to a `tasks.json` file, like:
```
        {
            "type": "shell",
            "label": "test:shell:task",
            "command": "sleep 2 && echo shell task was completed"
        },
```
An output for all cases described above should be displayed.
Please let me know if you have own use case and it doesn't work for you.

![fix_theia_tasks_output](https://user-images.githubusercontent.com/5676062/79745187-65de3c00-8310-11ea-8491-a4e73f838885.gif)



Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>
